fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -211,34 +218,86 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
             owner = viewLifecycleOwner,
             view = binding.root,
         )
+
+        setTranslationFragmentResultListener()
+    }
+
+    private fun setTranslationFragmentResultListener() {
+        setFragmentResultListener(
+            TRANSLATION_KEY,
+        ) { _, result ->
+            result.getString(TRANSLATION_STATUS_STATE_KEY)?.let { translationStatus ->
+                result.getString(SESSION_ID)?.let {
+                    if (it == getCurrentTab()?.id) {
+                        if (translationStatus == IN_PROGRESS) {
+                            FenixSnackbar.make(
+                                view = binding.dynamicSnackbarContainer,
+                                duration = Snackbar.LENGTH_SHORT,
+                                isDisplayedWithBrowserToolbar = true,
+                            )
+                                .setText(requireContext().getString(R.string.translation_in_progress_snackbar))
+                                .show()
+                        }
+
+                        val updateContentDescriptionSelected = result.getString(
+                            TranslationsDialogFragment.TRANSLATED_CONTENT_DESCRIPTION,
+                        )
+
+                        translationsAction?.setSelected(
+                            selected = true,
+                            notifyListener = false,
+                            updateContentDescriptionSelected = updateContentDescriptionSelected,
+                        )
+                    }
+                }
+            }
+        }
     }
 
     private fun initTranslationsAction(context: Context) {
         if (!context.settings().enableTranslations) {
             return
         }
+        if (translationsAction == null) {
```

Comment:
Will this button always be available as long as translations are enabled, even if the page is detected to be in the users default language?

Reply:
I don't know about this. The request was to show the translation toolbar button if the settings were enabled.

Reply:
I have some more information overall about the toolbar icon:
* If the page is detected to be in a user preferred language -> Do not show toolbar icon - change in Bug [1875817](https://bugzilla.mozilla.org/show_bug.cgi?id=1875817)
* If the page is detected to be in a user preferred language  -> User visits translations from the browser menu
   * If they translate: Show purple translated state toolbar icon
   * If they do not translate: No change, no icon on toolbar
   
Some more information in this [thread](https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD?node-id=7246:207835&mode=design#677883739) too (easily could have summarized wrong, etc.).

Reply:
Cool, thanks! We can always update the behavior in a follow-up but was just curious